File: FPG.Core/Factories/SaltGenerationAlgorithmFactory.cs
using System;

using FPG.Algorithms;
using FPG.Interfaces;

namespace FPG.Factories
{
    public static class SaltGenerationAlgorithmFactory
    {
        public static ISaltGenerationAlgorithm Create(String name)
        {
            switch (name.ToLower())
            {
                case "wp":
                    return new WPSaltGenerationAlgorithm();
                default:
                    throw new ArgumentException("Unknown algorithm name");
            };
        }
    }
}

File: lib/core/factories/SaltGenerationAlgorithmFactory.dart
import 'package:fpg/core/algorithms/saltgeneration/WPSaltGenerationAlgorithm.dart';
import 'package:fpg/core/interfaces/ISaltGenerationAlgorithm.dart';

class SaltGenerationAlgorithmFactory {
  static ISaltGenerationAlgorithm create(String name) {
    switch (name.toLowerCase()) {
      case "wp":
        return WPSaltGenerationAlgorithm();
      default:
        throw FormatException("Unknown algorithm name");
    }
  }
}
